utilities/transactions/write_unprepared_txn.h

```diff
@@ -153,6 +185,14 @@ class WriteUnpreparedTxn : public WritePreparedTxn {
   // locked for efficiency reasons. For recovered transactions, skip unlocking
   // keys when transaction ends.
   bool recovered_txn_;
+
+  // Track the largest sequence number at which we performed snapshot
+  // validation. If snapshot validation was skipped because no snapshot was set,
+  // then this is set to kMaxSequenceNumber. This value is useful because it
+  // means that for keys that have unprepared seqnos, we can guarantee that no
+  // committed keys by other transactions exist at largest_validated_seq_ <
+  // commit_seq.
+  SequenceNumber largest_validated_seq_;
```

Comment:
It seems that although still correct under rollback to save point, it might return error unnecessarily if the rollback has canceled out the write-without-snapshot issue. If that is correct, can we leave a comment here to have it documented.

Reply:
I'll leave a comment.